source/Hardpoint.cpp

```diff
@@ -201,13 +227,25 @@ void Hardpoint::Step()
 
 
 // Adjust this weapon's aim by the given amount, relative to its maximum
-// "turret turn" rate.
+// "turret turn" rate. Up to its angle limit.
 void Hardpoint::Aim(double amount)
 {
 	if(!outfit)
 		return;
 
-	angle += outfit->TurretTurn() * amount;
+	const double add = outfit->TurretTurn() * amount;
+	if(isOmnidirectional)
+		angle += add;
+	else
+	{
+		const Angle newAngle = angle + add;
+		if(add < 0. && minArc.IsInRange(newAngle, angle))
+			angle = minArc;
+		else if(add > 0. && maxArc.IsInRange(angle, newAngle))
+			angle = maxArc;
+		else
+			angle += add;
```

Comment:
I wonder if we could just do an angle::clamp function

Reply:
Looking at I thought there's a lot of similar structure in a few other place but when I actually check it's apparently not, this seems to be the only case so maybe we can just leave this for now?
Or if you (or anyone else) want me to make it now I can also do that.

Reply:
leave it for now ye, was just a thought i had